### PR TITLE
Promisify

### PR DESCRIPTION
**Pull request by Thaina**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,8 +45,18 @@ module.exports.setup = function (cb) {
 	], cb);
 };
 
-module.exports.validate = function (service, receipt, cb) {
-	switch (service) {
+module.exports.validate = function(service,receipt,cb) {
+	if(!cb && Promise)
+	{
```
Comment (voltrue2) [inline index.js]: Hi Thank you for the PR!
Would you mind matching the coding style with the rest here by doing `if (!cb && Promise) {` instead?

File: index.js [hunk 2]
```diff
@@ -35,8 +45,18 @@ module.exports.setup = function (cb) {
...
+		return new Promise(function(resolve,reject) {
+			self.validate(service,receipt,function(error,response) {
+				return error ? reject(error) : resolve(response);
+			});
+		});
+	}
+
+	switch(service)
```
Comment (voltrue2) [inline index.js]: Same request as above: `switch (service) {` please.

File: index.js [hunk 3]
```diff
@@ -50,12 +70,21 @@ module.exports.validate = function (service, receipt, cb) {
...
 		default:
-			return cb(new Error('invalid service given: ' + service));
+			cb(new Error('invalid service given: ' + service));
+			break;
 	}
 };
 
 module.exports.validateOnce = function (service, secretOrPubKey, receipt, cb) {
-	
+	if(!cb && Promise)
```
Comment (voltrue2) [inline index.js]: Same here.

File: index.js [hunk 4]
```diff
@@ -125,9 +152,16 @@ module.exports.getPurchaseData = function (purchaseData, options) {
 };
 
 module.exports.refreshGoogleToken = function (cb) {
-	
-	google.refreshToken(cb);
+	if(!cb && Promise)
```
Comment (voltrue2) [inline index.js]: Same here also.

**Comment by iferrera** [comment]
would u accept this PR?

**Comment by voltrue2** [comment]
Hello 

My apologies for not getting back to you on this PR.
I am trying to catch up on issues and PRs right now and my focus has been extending features and bug fixes so far. 

I will definitely get to this PR for sure!

Cheers

**Comment by iferrera** [comment]
glad to hear that! thanks!

**Comment by voltrue2** [comment]
So would you please go over all of my comments. I am pretty picky about coding style and I would like to keep the same style through out.

Cheers

**Comment by voltrue2** [comment]
Hi,

Thanks for the quick response to my request.
Would you do me one last favor in this?

`if(...)` -> `if (...)` etc. 

Bsasically I need a space between `if` and `(...)`, `switch` and `(...)`, `for` and `(...)`.
I know I am being picky, but I hope you bear with me on these points.

Cheers

**Comment by voltrue2** [comment]
Thank you!

**Comment by voltrue2** [comment]
As soon as I confirm with the Travis CI tests, I will merge the PR!
I am in Tokyo time zone, so earliest is for me tomorrow morning.

Thanks again!

File: index.js [hunk 5]
```diff
@@ -35,7 +44,15 @@ module.exports.setup = function (cb) {
 	], cb);
 };
 
-module.exports.validate = function (service, receipt, cb) {
+module.exports.validate = function(service,receipt,cb) {
```
Comment (voltrue2) [inline index.js]: `function(...` needs to be `function (...` please

File: index.js [hunk 6]
```diff
@@ -35,7 +44,15 @@ module.exports.setup = function (cb) {
...
+	if (!cb && Promise) {
+		return new Promise(function(resolve,reject) {
```
Comment (voltrue2) [inline index.js]: `function(...` needs to be `function (...` please

File: index.js [hunk 7]
```diff
@@ -35,7 +44,15 @@ module.exports.setup = function (cb) {
...
+			self.validate(service,receipt,function(error,response) {
```
Comment (voltrue2) [inline index.js]: `function(...` needs to be `function (...` please

File: index.js [hunk 8]
```diff
@@ -50,12 +67,20 @@ module.exports.validate = function (service, receipt, cb) {
...
-			return cb(new Error('invalid service given: ' + service));
+			cb(new Error('invalid service given: ' + service));
+			break;
 	}
 };
 
 module.exports.validateOnce = function (service, secretOrPubKey, receipt, cb) {
-	
+	if (!cb && Promise) {
+		return new Promise(function(resolve,reject) {
```
Comment (voltrue2) [inline index.js]: We need spaces where we need them. Please see my comments above.

File: index.js [hunk 9]
```diff
@@ -50,12 +67,20 @@ module.exports.validate = function (service, receipt, cb) {
...
+			self.validate(service,receipt,function(error,response) {
```
Comment (voltrue2) [inline index.js]: We need spaces where we need them. Please see my comments above.

File: index.js [hunk 10]
```diff
@@ -75,16 +100,14 @@ module.exports.validateOnce = function (service, secretOrPubKey, receipt, cb) {
 			amazon.validatePurchase(secretOrPubKey, receipt, cb);
 			break;
 		default:
-			verbose.log('<.validateOnce>', secretOrPubKey, receipt);
-			return cb(new Error('invalid service given: ' + service));
+			verbose.log('<.validateOnce>',secretOrPubKey,receipt);
```
Comment (voltrue2) [inline index.js]: We need spaces where we need them. Please see my comments above.

File: index.js [hunk 11]
```diff
@@ -125,9 +148,15 @@ module.exports.getPurchaseData = function (purchaseData, options) {
 };
 
 module.exports.refreshGoogleToken = function (cb) {
-	
-	google.refreshToken(cb);
+	if (!cb && Promise) {
+		return new Promise(function(resolve,reject) {
```
Comment (voltrue2) [inline index.js]: We need spaces where we need them. Please see my comments above.

File: index.js [hunk 12]
```diff
@@ -125,9 +148,15 @@ module.exports.getPurchaseData = function (purchaseData, options) {
...
+			google.refreshToken(function(error,response) {
```
Comment (voltrue2) [inline index.js]: We need spaces where we need them. Please see my comments above.

**Comment by Thaina** [comment]
It would be easier if you just pull it into your IDE and auto format it as your setting

Your formatting is not everyone format you know? I have my convention too and every of your request I need to fix it line by line

Seriously, convention should never be the reason to concern. Every IDE has autoformat for you

So I will stop caring on this thread now. I'm done

**Comment by Thaina** [comment]
As the last word

It's **YOUR** need, not **OUR** need

Our **NEED** is to use your library as `Promise` Not to concern about **YOUR** code convention. Because we never see it

**Comment by Thaina** [comment]
ps.
If you really care about **YOUR** convention that much you should write your travis to check for convention too. So I would know that I should not waste my time here

**Comment by nitrag** [comment]
Would like to see this feature. Bump

**Comment by voltrue2** [comment]
Closing as it is no longer relavent.

**Pull request closed without merging**